### ISSUE TITLE
Emic Automata

### ISSUE THREAD

**Issue by maetl** [NaNoGenMo/2017#123]
This is a book generated using elementary cellular automata with two randomly selected units of text mapped to the binary states of each cell.

The starting automaton is seeded with a randomly selected list of states. This seed is used to generate a wall of text—a study of repetition and contrast which combines the universal pattern generated by the automaton with the particular visual density of the selected units of text.

The number of cells in each generation and the number of generations are dependent on the length of the selected units. For this to work, the selected units must be the exact same length and the output text must be set in a monospace font.

![image](https://user-images.githubusercontent.com/24809/33316382-e1e1ce46-d487-11e7-80fe-818c6d4e0d63.png)

**Comment by greg-kennedy** [comment]
So, Conway's "Game of Life" with words?

**Comment by nossidge** [comment]
More like Wolfram's [256 binary rules](http://mathworld.wolfram.com/CellularAutomaton.html).

**Comment by maetl** [comment]
Yes, it’s based on the elementary (1D) automata studied extensively by Wolfram and others.

Game of Life is generally represented in 2D grid space with edge cells in either 4 or 8 directions. I haven’t figured out how to translate this 2D space to a text representation (yet!), but the 1D automata structure has a fairly intuitive mapping to the linear ordering of words in prose/book conventions.

**Comment by maetl** [comment]
One possible application of Game of Life (and other related automata) might be to treat the frames/generations as evolving a story in discrete steps, ie: chapters, passages, events. Although we’re really in the realm of concrete poetry/ascii art here, so not sure how far we can get with narrative structure, but it’s an interesting idea.

**Comment by maetl** [comment]
I’ve now published this — code is at [maetl/emic-automata](https://github.com/maetl/emic-automata) and some example PDFs are included above. It’s pretty weird.

**Comment by maetl** [comment]
Update:

- [Bundle of 100 generated books available for download](https://github.com/maetl/emic-automata)
- [Notes on the concept and method behind this](https://maetl.net/notes/storyboard/generative-writing-with-cellular-automata)
